hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -266,16 +295,23 @@ public AbfsRestOperation createPath(final String path, final boolean isFile, fin
     return op;
   }
 
-  public AbfsRestOperation renamePath(final String source, final String destination, final String continuation)
+  public AbfsRestOperation renamePath(String source, final String destination, final String continuation)
           throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
 
-    final String encodedRenameSource = urlEncode(FORWARD_SLASH + this.getFileSystem() + source);
+    final AbfsUriQueryBuilder srcQueryBuilder = new AbfsUriQueryBuilder();
+    appendSASTokenToQuery(source, SASTokenProvider.RENAME_SOURCE_OPERATION, srcQueryBuilder);
+    String sasToken = srcQueryBuilder.toString();
+
+    final String encodedRenameSource =
+        urlEncode(FORWARD_SLASH + this.getFileSystem() + source) + sasToken;
+    LOG.trace("Rename source queryparam added {}", encodedRenameSource);
```

Comment:
We should only create the builder and append the sasToken when authType is AuthType.SAS.  I don't think we should create throw-away objects.  The driver should be lean and only allocate when necessary to reduce overall pressure on GC.

Reply:
Are you suggesting we move the AuthType == SAS check back to the API methods ? It was moved to appendSASTokenToQuery() as it was a repeating code across API methods.

Reply:
As per the recent Thomas's comment, will continue with the Auth check within appendSASTokenToQuery() method.